evap/staff/views.py

```diff
@@ -909,27 +910,14 @@ def semester_preparation_reminder(request, semester_id):
 def semester_grade_reminder(request, semester_id):
     semester = get_object_or_404(Semester, id=semester_id)
 
-    courses = (
-        semester.courses.filter(
-            evaluations__state__gte=Evaluation.State.EVALUATED,
-            evaluations__wait_for_grade_upload_before_publishing=True,
-            gets_no_grade_documents=False,
-        )
-        .distinct()
-        .prefetch_related("responsibles")
-    )
-
-    courses = [course for course in courses if not course.final_grade_documents.exists()]
-    courses.sort(key=lambda course: course.name)
-
-    responsibles = UserProfile.objects.filter(courses_responsible_for__in=courses).distinct()
+    courses = Course.objects_with_missing_final_grades().filter(semester=semester).prefetch_related("responsibles")
+    courses = sorted(courses, key=lambda course: course.name)
```

Comment:
Not sure if we care about this here, but we _could_ go ahead and add typing to this function. This would require the `courses` variable to not be reassigned with a different type here. Doesn't really matter for the feature though